probe/tag/tagger.go

```diff
@@ -0,0 +1,25 @@
+package tag
+
+import "github.com/weaveworks/scope/report"
+
+// Tagger tags nodes with value-add node metadata.
+type Tagger interface {
+	Tag(r report.Report, ts report.TopologySelector, id string) report.NodeMetadata
```

Comment:
Maybe it would be helpful to document all the topologies you want to introduce (even if they are not in this PR)?  I get endpoint, container & process.  Whats address?

I have yet thought of what I'm driving at, but it does strike me at the very least that:

```
	myNodeMetadata, ok := ts(r).NodeMetadatas[id]
	if !ok {
		return report.NodeMetadata{}
	}
```

Appears at the top of ~every implementation, and we're calling these taggers on topologies where we expect them to fail - making it hard to distinguish programmer errors from normal operation.  I don't like how things fail silently.

I'm having a little play to see if I can come up with something nicer.